### PR TITLE
remediations from orijtech audit

### LINKED ISSUE #180: x/airdrop: redundant code in InitGenesis by placing zero value into map[string]uint64:: low priority
This code https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/airdrop/genesis.go#L20-L22 is redundant and unnecessary given that in Go, the zero value of primitive types is useful so that code can simply be deleted and
https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/airdrop/genesis.go#L23 will simply work and accumulate with the zero value and this is an exhibit https://go.dev/play/p/3aXlcTHIuIn

```go
package main

func main() {
	// Simplified version.
	m := make(map[string]uint64)
	m["e"] += 2
	println(m["e"])

	// Redundant version.
	n := make(map[string]uint64)
	if _, ok := n["e"]; !ok {
		n["e"] = 0
	}
	n["e"] += 2
	println(n["e"])
}
```

which produces
```shell
2
2
```

### LINKED ISSUE #182: x/interchainstaking/types: inconsistent error handling in ConvertMemoToOrdinalIntents
Missing error check when converting memo to valoper address:

https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/interchainstaking/types/zones.go#L115

Perhaps addresses are always valid by higher level logic, but then earlier error checks are redundant:

https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/interchainstaking/types/zones.go#L98-L107

CC @odeke-em

### LINKED ISSUE #184: x/interchainstaking/keeper: possible missing return in Keeper.BeginBlocker error path
In `Keeper.BeingBlocker`

https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/interchainstaking/keeper/abci.go#L23

there is a panic commented out and a comment explaining why. However, there is no `return` statement or similar to halt further execution of the method. If that is intentional, I suggest a comment explaining why continuing is OK.

Same comment applies to the next error path,

https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/interchainstaking/keeper/abci.go#L25-L27

where a missing return may be OK because it completes the `blockheight%10` block.

CC @odeke-em

### LINKED ISSUE #187: x/interchainstaking/keeper: possible missing out-of-bounds check in AccountBalanceCallback
There's no length check before slicing:

https://github.com/ingenuity-build/quicksilver/blob/a7f142ab99e978482f66248d0009b90820d0f0ce/x/interchainstaking/keeper/callbacks.go#L432

There may be a higher-level reason that `len(request.Query)` is always greater than zero.

CC @odeke-em

### PR DESCRIPTION
**Pull request by joe-bowman**
- add comments explaining the non-return under error conditions in x/interchainstaking beginBlocker; fixes #184
- remove redundant code; fixes #180
- refactor, test and bug fix ParseDelegationKey #181
- improved error handling; closes #182 #183
- improved input validation; closes #187

### CODE REVIEW

File: x/interchainstaking/keeper/msg_server.go [hunk 1]
```diff
@@ -169,7 +172,7 @@ func (k msgServer) RequestRedemption(goCtx context.Context, msg *types.MsgReques
 	for _, delegator := range delegators {
 		icaAccount, err := zone.GetDelegationAccountByAddress(delegator)
 		if err != nil {
-			panic(err) // panic here because something is terribly wrong if we can't find the delegation bucket here!!!
+			return nil, err // panic here because something is terribly wrong if we can't find the delegation bucket here!!!
```
Comment (zanicar) [inline x/interchainstaking/keeper/msg_server.go:175]: maybe we should keep the panic here?

**Review by zanicar — COMMENTED** [review]
just the one panic we should maybe keep...

File: x/interchainstaking/keeper/msg_server.go (hunk 1 above)
Comment (joe-bowman) [reply x/interchainstaking/keeper/msg_server.go]: Done

**Review by zanicar — APPROVED** [review]
lgtm

**Pull request merged**